curriculum/challenges/english/25-front-end-development/lecture-working-with-the-dom-click-events-and-web-apis/673368ccf52205329b729378.md

```diff
@@ -10,6 +10,140 @@ dashedName: what-is-the-difference-between-innertext-textcontent-and-innerhtml
+</div>
+```
+
+If we get a reference to this HTML element in our JavaScript code using `getElementById()`, we can access the `innerText` property of this element.
+
+```js
+const container = document.getElementById("container");
+console.log(container.innerText);
+```
+
+This is the inner text of this element.
+
+```md
+Hello, World!
+I'm learning JavaScript
+```
+
+The property returns a string with the text contained within the element, including text from its descendants.
+
+You should know that `innerText` only returns the text that is visible at the particular moment when the string is requested. If a child element is hidden, its text won't be visible.
+
+This is an example where the second paragraph is hidden.
+
+```html
+<div id="container">
+  <p>Hello, World!</p>
+  <p hidden>I'm learning JavaScript</p>
+</div>
+```
+
+If we try to log the `innerText` again, now the output won’t have the text of the second paragraph.
+
+```js
+console.log(container.innerText);
+```
+
+This will be the output:
+
+```md
+Hello, World!
+```
+
+You can set the `innerText` of an HTML element like this, but this will replace the existing text and add a line break element (`br`) element for every line break.
+
+```js
+container.innerText = "JavaScript is awesome!";
+```
+
+Since `innerText` takes visibility into account, getting its value triggers a process called "reflow", that recalculates the position of certain elements on the website. This process can be computationally intensive, so you should avoid triggering it if possible.
+
+Great. Now let's talk about `textContent`. 
+
+`textContent` returns the plain text content of an element, including all the text within its descendants.
+
+The most important difference between `innerText` and `textContent` is that `textContent` always returns the full text content of an HTML element and its descendants, regardless of whether it's visible or hidden.
+
+Here we have the same example in HTML:
+
+```html
+<div id="container">
+  <p>Hello, World!</p>
+  <p>I'm learning JavaScript</p>
+</div>
+```
+
+If you try to access this property, you'll see the text of the element and its descendants as the output, keeping the indentation and spacing.
+
+```js
+const container = document.getElementById("container");
+console.log(container.textContent);
+```
+
+```md
+Hello, World!
+I'm learning JavaScript
+``` 
+
+If an HTML element is not visible, like you can see over here, where we’ve hidden the second paragraph, its text will still be included in this property.
+
+```html
+<div id="container">
+  <p>Hello, World!</p>
+  <p hidden>I'm learning JavaScript</p>
+</div>
+```
+
+You will see the same output:
+
+```md
+Hello, World!
+I'm learning JavaScript
+``` 
+  
+`textContent` will also include the content of elements like `script` and `style`.
+
+If you try to replace the value of `textContent` on a node, it will remove all its child nodes and replace them with a single text node containing the new string.
```

Comment:
```suggestion
If you try to replace the value of `textContent` on a node, it will remove all its child nodes and replace them with a single text node containing the new string:
```